src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -140,7 +140,7 @@ private HttpEnvironmentProxy(Uri httpProxy, Uri httpsProxy, string bypassList)
             if (!string.IsNullOrWhiteSpace(bypassList))
             {
                 string[] list = bypassList.Split(',');
-                List<string> tmpList = new List<string>();
+                List<string> tmpList = new List<string>(list.Length);
```

Comment:
Even though we're only allocating one of these proxies per handler, it'd still be nice to reduce allocations involved.  I'm fine with it being a follow-up effort to not block this PR, but there could be a number of allocations happening here if a bypassList is provided:
- Allocating the string[] for the split
- Allocating one string per entry in the bypass list
- Allocating the List for the resulting strings
- Allocating the one or more arrays backing the List
- Allocating a trimmed string if there was any whitespace that needed to be removed
- Allocating the final array

Some of these can fairly easily be avoided, e.g. by using https://github.com/dotnet/corefx/blob/master/src/Common/src/System/IO/StringParser.cs to do the split, by using Trim on a span to avoid intermediate string allocations if there is whitespace to be removed, by using, etc.

Again, not needed for this PR, but something to keep in mind.  It's probably not needed for 2.1, either.

Reply:
would it make sense to use Span to create Uri? Even the split is just convenience and fining ';' should be easy enough.

Reply:
I did some re-work inspired but the linked function. Since I want to do more than just the split I could not use it directly. BTW if that was inspired by strcspn() it would be nice if one could pass in list of white spaces instead of just ' '.